### ISSUE TITLE
External window, created with Application.Current.OpenWindow crashes the app on close

### ISSUE THREAD

**Issue by alexeystrakh** [dotnet/maui#3984]
### Description

I'm using the multi-window support feature `Application.Current.OpenWindow` to create an external window for my MAUI app. And if I drop regular content there it works just fine. But if you put a WebView there (on the external page), and try to close the page, the app will crash.

I can reproduce it on the Windows platform only, MacCatalyst works fine.
And I can reproduce it with WebView only, no other content generates a similar crash for me.
Also if you put an empty WebView (without loading a web page using the Source), it will work either, no crash in this case.

### Steps to Reproduce

1. Create a new MAUI app in VS2022 17.1 P2
2. Drop a button and on click execute the following code:
```
var page = new ContentPage() { Content = new WebView { Source = "https://google.com" } };
var window = new Window(page);
Application.Current.OpenWindow(window);
```
3. Close the opened windows with the cross in the top right corner

Expected: the external window is closed
Actual: the app crashes with the following unhandled exception (track trace is null, no additional details) in the generated code of `App.g.i.cs.InitializeComponent`:

```
"Microsoft.UI.Xaml.UnhandledExceptionEventArgs"
e.Exception.ToString()
"System.ObjectDisposedException: The object has been closed. (0x80000013)"
```

![image](https://user-images.githubusercontent.com/3697084/148461403-1aa830f6-6570-44d1-8da6-2d3036ea7689.png)

### Version with bug

Preview 11

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10.0.19041

### Did you find any workaround?

No workaround was found

### Relevant log output

```shell
Output:

The program '[9292] xxx.exe' has exited with code 3221226107 (0xc000027b).
```

Exception details:
```
"Microsoft.UI.Xaml.UnhandledExceptionEventArgs"
e.Exception.ToString()
"System.ObjectDisposedException: The object has been closed. (0x80000013)"
```

EventViewer:
```
Faulting application name: xxx.exe, version: 1.0.0.0, time stamp: 0x61954b73
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1387, time stamp: 0x0b9a844a
Exception code: 0xc000027b
Fault offset: 0x000000000010b302
Faulting process id: 0x244c
Faulting application start time: 0x01d8034e06ca0fe9
Faulting application path: xxx\bin\Debug\net6.0-windows10.0.19041\win-x64\AppX\xxx.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: de259b8e-4134-44d4-8d99-6b88956fc813
Faulting package full name: 0A8834C6-F23D-4CEB-86B5-B57FD5D15D23_1.0.0.0_neutral__9zz4h110yvjzm
Faulting package-relative application ID: App
```
```

**Comment by mattscheffer** [comment]
Repros on 17.2.0 Preview 1. Tested on Android 11 and Windows. In android, the main app closes instantly when the google page is closed. On Windows however it is currently crashing upon creating the new window. 

Repro project is here: https://microsoft.sharepoint.com/:u:/r/teams/Mauitriage/Shared%20Documents/General/ReproProjects/3984.zip?csf=1&web=1&e=gYVD4M

**Comment by Redth** [comment]
This is a crash from the WinAppSDK that should be fixed in 1.0.1 of their nuget and MAUI's Preview 14.
